lib/matplotlib/backends/qt_editor/figureoptions.py

```diff
@@ -235,15 +235,26 @@ def apply_callback(data):
 
         # re-generate legend, if checkbox is checked
         if generate_legend:
-            draggable = None
-            ncol = 1
+            kwargs = {'ncol' : 1}
             if axes.legend_ is not None:
                 old_legend = axes.get_legend()
-                draggable = old_legend._draggable is not None
-                ncol = old_legend._ncol
-            new_legend = axes.legend(ncol=ncol)
-            if new_legend:
-                new_legend.set_draggable(draggable)
+                kwargs = {name: getattr(old_legend, name) for name in
+                                ["numpoints", "markerscale", "shadow",
+                                 "columnspacing", "scatterpoints",
+                                 "handleheight", "borderpad",
+                                 "labelspacing", "handlelength",
+                                 "handletextpad", "borderaxespad"]}
+
+                kwargs['ncol'] = old_legend._ncol
+                kwargs['loc'] = old_legend._get_loc()
+                kwargs['mode'] = old_legend._mode
+                kwargs['title'] = old_legend.get_title().get_text()
+                kwargs['frameon'] = old_legend.get_frame_on()
+                kwargs['bbox_to_anchor'] = old_legend._bbox
+                kwargs['handler_map'] = old_legend.get_legend_handler_map()
+
+            handles, labels = axes.get_legend_handles_labels()
```

Comment:
Can we do

```python
new_legend = axes.legend(**kwargs)
```

instead?

Reply:
Yeah sure, that works! will raise a patch.